rpc/context.go

```diff
@@ -45,15 +46,17 @@ const (
 // service.
 func NewServer(ctx *Context) *grpc.Server {
 	var s *grpc.Server
-	if ctx.Insecure {
-		s = grpc.NewServer()
-	} else {
+	opts := []grpc.ServerOption{
+		grpc.MaxMsgSize(math.MaxInt32), // TODO(peter,tamird): need tests before lowering
+	}
+	if !ctx.Insecure {
 		tlsConfig, err := ctx.GetServerTLSConfig()
 		if err != nil {
 			panic(err)
 		}
-		s = grpc.NewServer(grpc.Creds(credentials.NewTLS(tlsConfig)))
+		opts = append(opts, grpc.Creds(credentials.NewTLS(tlsConfig)))
 	}
+	s = grpc.NewServer(opts...)
```

Comment:
Also worth noting that once #8265 is in, we'll catch this type of issue in
our nightly load tests. However, it'll still require more diagnosis than a
more specific test.

On Tue, Aug 2, 2016 at 9:56 PM Tamir Duberstein <notifications@github.com>
wrote:

> In rpc/context.go
> <https://github.com/cockroachdb/cockroach/pull/8274#discussion_r73269718>:
>
> >  	}
> > +	s = grpc.NewServer(opts...)
>
> let's file an issue about those tests
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8274/files/b0c512eb4fd9acc0c9b202949d967249f82137e1#r73269718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABffpvQs_vOz7XA_UIm0lx78LHMFU1xsks5qb_U_gaJpZM4JbMUi>
> .
>